### DISCUSSION TITLE
Having on_event("startup") run only once when using multiple workers

### DISCUSSION THREAD

**Discussion by JohnnyRacer** [fastapi/fastapi#9766] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from awesome_library import some_startup_func

app = FastAPI()

@app.get('/')
def test_get():
   return "Hello world"

@app.on_event("startup")
def startup_exec():
    some_startup_func() # Call some startup function
```

### Description

Hello, I am trying to run a startup function once when using multiple workers with Uvicorn with the `--workers n` whre `n` is the number of workers. It seems like the function actualy gets run `n` times, each with when a worker is started. Is there a way to change this behavior so that the function gets run only once instead `n` times?

The command I am using to start FastAPI:
```py
uvicorn app.main:app --workers 4
```

### Operating System

Linux, Windows, macOS

### FastAPI Version

0.98.0

### Python Version

3.8.16

**Comment by abhint** [comment, marked as answer]
#6209 It will help you.

**Reply by JohnnyRacer** [reply]
Thanks, it was what I needed.
